vite-project/src/components/Modal/AddLinkModal.jsx

```diff
@@ -0,0 +1,63 @@
+import { useState } from "react";
+import "../../style/AddLinkModal.css";
+
+function AddLinkModal({ onClose, cardListMenuData }) {
+  const [selectedFolder, setSelectedFolder] = useState("");
+
+  const handleFolderSelect = folderName => {
+    setSelectedFolder(folderName);
+  };
+
+  return (
+    <div className="addlinkmodal-section">
+      <div className="addlinkmodal-title-section">
+        <div className="addlinkmodal-maintitle">폴더에 추가</div>
+        <div className="addlinkmodal-subtitle">링크 주소</div>
+      </div>
+      <div className="addlinkmodal-folder-list">
+        {cardListMenuData?.data?.map(folder => (
+          <button
+            key={folder.id}
+            className={`addlinkmodal-folder-list-btns ${
+              selectedFolder === folder.name ? "selected" : ""
+            }`}
+            type="button"
+            onClick={() => handleFolderSelect(folder.name)}
+          >
+            <div
+              className={`addlinkmodal-folder-list-btn-name ${
+                selectedFolder === folder.name ? "selected" : ""
+              }`}
+            >
+              {folder.name}
+            </div>
+            {typeof folder.link === "object" &&
+              typeof folder.link.count === "number" && (
+                <div className="addlinkmodal-folder-list-btn-count">
+                  {folder.link.count}개 링크
+                </div>
+              )}
```

Comment:
folder가 link가 있고, count 속성이 숫자인지 확인하는 방식을 좀 더 간략히 작성할 수 있을거같아요
```suggestion
              <div className="addlinkmodal-folder-list-btn-count">
                {folder?.link?.count}개 링크
              </div>
```